### PR TITLE
ref: cache volta, node, yarn in github actions

### PR DESCRIPTION
**Pull request by asottile-sentry**
this aims to solve the flakiness described in https://getsentry.atlassian.net/browse/DEVINFRA-55

I don't expect this to significantly speed up the node setup, mostly just to make it more reliable -- notably I expect github's caches to be more reliable than always downloading all the time from volta

on the perf perspective: 

anecdotal -- but hopefully enough to demonstrate that there isn't a performance regression:

[before (cache hit)](https://github.com/getsentry/sentry/runs/7133395320?check_suite_focus=true):
- volta-cli/action: 1s
- cache dir: 2s  (this installs node/yarn)
- actions/cache@v3: 8s
- install dependencies: 33s

(**setup time: 11s**)

[after (cache miss)](https://github.com/getsentry/sentry/runs/7135058976?check_suite_focus=true):
- setup-volta: 5s
- install dependencies: 55s

(**setup time: 5s, cache miss time ~25s**)

[after (cache hit)](https://github.com/getsentry/sentry/runs/7135059153?check_suite_focus=true):
- setup-volta: 11s
- install dependencies: 31s

(**setup time: 11s**)

### CODE REVIEW

File: .github/actions/setup-volta/bin/setup-volta [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import io
+import json
+import os.path
+import secrets
+import subprocess
+import sys
+import tarfile
+import urllib.request
+
+VOLTA = "1.0.8"
```
Comment (joshuarli) [inline .github/actions/setup-volta/bin/setup-volta:13]:
```suggestion
VOLTA_VERSION = "1.0.8"
```

Just to match `docker/builder.dockerfile`.

Comment (joshuarli) [reply .github/actions/setup-volta/bin/setup-volta:13]: Actually, hold on, I have an idea.

**Review by joshuarli — APPROVED** [review]
Nice.

I was thinking the downloading parts could be moved to `tools/` and reused everywhere, but it would all be obsolete as soon as redist. dev environments are ready. Right?

**Comment by asottile-sentry** [comment]
> > Nice.
> > I was thinking the downloading parts could be moved to `tools/` and reused everywhere, but it would all be obsolete as soon as redist. dev environments are ready. Right?
> 
> yeah at some point I would move this out and have it support the local machine (right now it does stuff which is specific to github actions which would need to be adjusted slash wouldn't be needed in the local case)

plus we'd want to add arm64 support

**Pull request merged**